pkg/util/docker_utils.go

```diff
@@ -49,56 +51,73 @@ func NewClient() (*client.Client, error) {
 	return cli, nil
 }
 
-func getLayersFromManifest(manifestPath string) ([]string, error) {
+func getLayersFromManifest(r io.Reader) ([]string, error) {
 	type Manifest struct {
 		Layers []string
 	}
 
-	manifestJSON, err := ioutil.ReadFile(manifestPath)
+	manifestJSON, err := ioutil.ReadAll(r)
 	if err != nil {
-		errMsg := fmt.Sprintf("Could not open manifest to get layer order: %s", err)
-		return []string{}, errors.New(errMsg)
+		return nil, err
 	}
 
 	var imageManifest []Manifest
-	err = json.Unmarshal(manifestJSON, &imageManifest)
-	if err != nil {
-		errMsg := fmt.Sprintf("Could not unmarshal manifest to get layer order: %s", err)
-		return []string{}, errors.New(errMsg)
+	if err := json.Unmarshal(manifestJSON, &imageManifest); err != nil {
+		return []string{}, fmt.Errorf("Could not unmarshal manifest to get layer order: %s", err)
 	}
 	return imageManifest[0].Layers, nil
 }
 
 func unpackDockerSave(tarPath string, target string) error {
 	if _, ok := os.Stat(target); ok != nil {
-		os.MkdirAll(target, 0777)
+		os.MkdirAll(target, 0644)
```

Comment:
Hmm, looks like you're getting permission errors on the test files now. I think you need to increase this to 666?